integrated_channels/canvas/models.py

```diff
@@ -89,13 +89,15 @@ class CanvasEnterpriseCustomerConfiguration(EnterpriseCustomerPluginConfiguratio
 
     canvas_company_id = models.CharField(
         max_length=255,
+        null=True,
         blank=True,
```

Comment:
Should this allow blank though? When ECS fills this out I imagine this field would be required no? Again, having null here means database-wise there will be no issues with not having current values, so you don't also need blank in that regard. The question is whether you want to allow someone to fill out the info for a canvas integration and *not* provide a company id.

Reply:
yeah that is a good question. not sure either. I am not actually sure how we are going to use this field, if at all. @alex-sheehan-edx may have thoughts here based on what has come out of oauth? and if we need this field for oauth and other purposes, looks like we must not allow blank or null either

Reply:
or at least not allow blank

Reply:
Slightly confused as to what the company ID represents here.. For OAuth, as of right now, the values I need are `canvas_base_url`, `client_id`, `client_secret`, `(client_?)code`, `refresh_token` (retrieved when the user is prompted by Canvas to authorize and clicks the confirmation button), `access_token` (retrieved by the previous values when instantiating the client's session) and the constants `oauth_token_auth_path` ("login/oauth2/token") and `oauth_redirect_uri` (a to be added endpoint for after the Canvas prompted authorization flow)

Reply:
yeah looks like may not need it here then. degreed was different, it uses it like this https://github.com/edx/edx-enterprise/blob/af5b13a109f611375c3fffbf97aa236cf0fd38c7/integrated_channels/degreed/transmitters/content_metadata.py#L27 but for canvas if this is never going to be used, then let's drop it?

Reply:
same with `provider_id`?

Reply:
I think provider_id is safe to get rid of for now. I'd however change canvas_company_id to canvas_account_number. We need that for a bunch of URLs, so we do need to store it per customer

Reply:
is `canvas_account_number` the :account_id in urls? so then it should be an integer?